docs/tokencache.md

```diff
@@ -2,16 +2,350 @@
 
 ## Introduction
 
-> PENDING
+The token cache implementation for MSAL, ADAL and Android broker is found in the Android Common Library.  
 
-## Class Diagram
+## References 
 
-> PENDING
+- Universal Cache Schema
 
-## Interaction Diagram
+## Universal cache and protocol response handling
 
-> PENDING
+When looking at the token cache there are really 2 levels to it.  The lower level is based on the universal cache schema referenced above.  The higher level deals with the result of a particular protocol request and mapping that to the universal schema.  
+
+So in effect we get an oAuth 2 Token Response, which in the case of MSAL includes an ID Token and an Access Token and we translate:
+
+ID Token & Client_Info -> Account & Tenant Profile
+Access Token -> Credential
+Refresh Token -> Credential
+
+## Universal Schema at a high level / OAuth v2 Mapping
+
+- Accounts
+  - ID Token of account home tenant
+- Tenant Profiles - The representation of 1 account in 1 or more tenants (external account guest/member scenarios)
+  - ID Token of the account home tenant
+  - ID Token from other tenants that the account is a member of
+- Credentials
+  - Id Token
+  - Access Tokens
+  - Refresh Tokens
+  - etc....
+
+### Relationships
+
+1 Account has 0 or more credentials
```

Comment:
```suggestion
1 Account has 1 or more credentials (minimum, the `AccountRecord` + `RefreshTokenRecord`)
```

For the purposes of how calls like `getAccounts()` works, [the refresh token must be present](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common4j/src/main/com/microsoft/identity/common/java/cache/MsalOAuth2TokenCache.java#L1193-L1198) for us to consider the account "available" -- this is to ensure we don't wind up in a situation where `getAccounts()` returns an `AccountRecord` for which no RT exists (meaning we couldn't do any silent auth)